redis-om-spring/src/main/java/com/redis/om/spring/repository/query/RediSearchQuery.java

```diff
@@ -266,44 +270,54 @@ public QueryMethod getQueryMethod() {
     return queryMethod;
   }
 
-  public boolean isAnnotationBased() {
-    return annotationBased;
-  }
-
   private Object executeQuery(Object[] parameters) {
     SearchOperations<String> ops = modulesOperations.opsForSearch(searchIndex);
-    Query query = new Query(prepareQuery(parameters));
+    String preparedQuery = prepareQuery(parameters);
+    Query query = new Query(preparedQuery);
     query.returnFields(returnFields);
 
     Optional<Pageable> maybePageable = Optional.empty();
 
     if (queryMethod.isPageQuery()) {
-      maybePageable = Arrays.stream(parameters)
-          .filter(o -> o instanceof Pageable)
-          .map(o -> (Pageable) o)
-          .findFirst();
+      maybePageable = Arrays.stream(parameters).filter(o -> o instanceof Pageable).map(o -> (Pageable) o).findFirst();
 
       if (maybePageable.isPresent()) {
         Pageable pageable = maybePageable.get();
-        query.limit(Long.valueOf(pageable.getOffset()).intValue(), pageable.getPageSize());
-
-        if (pageable.getSort() != null) {
-          for (Order order : pageable.getSort()) {
-            query.setSortBy(order.getProperty(),  order.isAscending());
+        if (!pageable.isUnpaged()) {
+          query.limit(Long.valueOf(pageable.getOffset()).intValue(), pageable.getPageSize());
+  
+          if (pageable.getSort() != null) {
```

Comment:
What about not paged, but sortable? Or did I misunderstand (not unlikely)

Reply:
The is support for that too, the pattern in Spring Data seems to pass a Pageable that has defaults for the paging, e.g. 0 to Collection.size and include the Sort object as needed